lib/active_record_inherit_assoc.rb

```diff
@@ -29,9 +31,11 @@ def target_scope
 
   def attribute_inheritance_hash
     return nil unless reflection.options[:inherit]
+    inherit_allowed_list = reflection.options[:inherit_allowed_list]
 
     Array(reflection.options[:inherit]).each_with_object({}) do |association, hash|
       assoc_value = owner.send(association)
+      assoc_value = [assoc_value, inherit_allowed_list].flatten if inherit_allowed_list
```

Comment:
1 level of flattening should be enough ... also makes it faster
```suggestion
      assoc_value = [assoc_value, inherit_allowed_list].flatten(1) if inherit_allowed_list
```

even better:
```
assoc_value = Array(assoc_value) + inherit_allowed_list if inherit_allowed_list
```